miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java

```diff
@@ -172,6 +180,27 @@ public PoolDto updatePool(@PathVariable Long poolId, @RequestBody PoolDto pool)
     return Dtos.asDto(poolService.get(poolId), true);
   }
 
+  @RequestMapping(value = "/{poolId}/assign", method = RequestMethod.POST, produces = "application/json")
+  @ResponseStatus(code = HttpStatus.OK)
+  public void assignPool(@PathVariable Long poolId, @RequestBody List<Long> partitionIds) throws IOException {
+    Pool pool = poolId == 0 ? null : poolService.get(poolId);
+    partitionIds.stream()//
+        .map(WhineyFunction.rethrow(containerService::getPartition))//
+        .peek(partition -> {
+          if (pool != null && partition.getSequencerPartitionContainer().getPlatform().getPlatformType() != pool.getPlatformType()) {
+            throw new RestException(
+                String.format("%s %d in %s is not compatible with pool %s.",
+                    partition.getSequencerPartitionContainer().getPlatform().getPlatformType().getPartitionName(),
+                    partition.getPartitionNumber(), partition.getSequencerPartitionContainer().getIdentificationBarcode(), pool.getName()),
+                Status.BAD_REQUEST);
+          }
+          partition.setPool(pool);
+        })//
+        .forEach(WhineyConsumer.rethrow(containerService::update));
+    pool.setReadyToRun(false);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "pool" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)